File: 프로그래머스/unrated/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution (String my_string, int[][] queries) {
        StringBuffer sb = new StringBuffer(my_string);
        for (int[] q : queries)
        {
            for (int i = 0; q[0] + i < q[1] - i; i++)
            {
                char ch = sb.charAt(q[0] + i);
                sb.setCharAt(q[0] + i, sb.charAt(q[1] - i));
                sb.setCharAt(q[1] - i, ch);                
            }
        }
        return sb.toString();
    }
}

File: 프로그래머스/Lv.0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.js
function solution(my_string, queries) {
    var arr = my_string.split('');
    for (const q of queries) {
        for (var i = 0; q[0] + i < q[1] - i; i++) {
            var tmp = arr[q[0] + i];
            arr[q[0] + i] = arr[q[1] - i];
            arr[q[1] - i] = tmp;
        }
    }
    return arr.join('');
}